modules/api1/controllers/ProjectController.php

```diff
@@ -102,6 +106,27 @@ public function actionVote($id)
             'votingResult' => $project->votingResult
         ];
     }
+
+    /**
+     * @param int $id
+     *
+     * @throws ServerErrorHttpException
+     * @throws UnauthorizedHttpException
+     */
+    public function actionDelete($id)
+    {
+        $user = $this->getCurrentUser();
+        if (!$user) {
+            throw new UnauthorizedHttpException('User should be authorized in order to manage project.');
+        }
+        
+        $project = $this->findProject($id, $user);
```

Comment:
I think the following check is missing:

```php
if (!UserPermissions::canManageProject($project)) {
    throw new ForbiddenHttpException(Yii::t('project', 'You can not delete this project.'));
}
```

Reply:
I add the condition. But still as an additional check, the code below only skips the owner.
````
protected function findProject($projectId, User $user)
    {
        /** @var Project $project */
        $project = Project::find()
            ->where(['id' => $projectId])
            ->hasUser($user)
            ->available()
            ->one();

        if ($project) {
            return $project;
        }
        
        throw new NotFoundHttpException("The requested project does not exist.");
    }
```

Reply:
Ah. Alright.